tonuino.ino

```diff
@@ -389,73 +445,79 @@ void playNextTrack(uint16_t globalTrack, bool directionForward, bool isInteracti
-  for (uint8_t i = 0; i < 6; i++) mifareKey.keyByte[i] = 0xFF;
+  for (uint8_t i = 0; i < 6; i++)
+    mifareKey.keyByte[i] = 0xFF;
 
   // check if card type is supported
   mifareType = mfrc522.PICC_GetType(mfrc522.uid.sak);
-  if (mifareType != MFRC522::PICC_TYPE_MIFARE_MINI && mifareType != MFRC522::PICC_TYPE_MIFARE_1K && mifareType != MFRC522::PICC_TYPE_MIFARE_4K) {
+  if (mifareType != MFRC522::PICC_TYPE_MIFARE_MINI && mifareType != MFRC522::PICC_TYPE_MIFARE_1K && mifareType != MFRC522::PICC_TYPE_MIFARE_4K)
+  {
     mfrc522.PICC_HaltA();
     mfrc522.PCD_StopCrypto1();
     returnCode = 255;
   }
-  else {
+  else
+  {
     // check if we can authenticate with mifareKey
-    mifareStatus = (MFRC522::StatusCode) mfrc522.PCD_Authenticate(MFRC522::PICC_CMD_MF_AUTH_KEY_A, mifareBlock, &mifareKey, &(mfrc522.uid));
-    if (mifareStatus != MFRC522::STATUS_OK) {
+    mifareStatus = (MFRC522::StatusCode)mfrc522.PCD_Authenticate(MFRC522::PICC_CMD_MF_AUTH_KEY_A, mifareBlock, &mifareKey, &(mfrc522.uid));
+    if (mifareStatus != MFRC522::STATUS_OK)
+    {
       mfrc522.PICC_HaltA();
       mfrc522.PCD_StopCrypto1();
       returnCode = 254;
     }
-    else {
+    else
+    {
       // read data from nfc tag
-      mifareStatus = (MFRC522::StatusCode) mfrc522.MIFARE_Read(mifareBlock, mifareData, &mifareDataSize);
-      if (mifareStatus != MFRC522::STATUS_OK) {
+      mifareStatus = (MFRC522::StatusCode)mfrc522.MIFARE_Read(mifareBlock, mifareData, &mifareDataSize);
+      if (mifareStatus != MFRC522::STATUS_OK)
+      {
         mfrc522.PICC_HaltA();
         mfrc522.PCD_StopCrypto1();
         returnCode = 253;
       }
-      else {
-        // for debug purposes, print the first 16 bytes of sector 1 / block 4
-        Serial.print(F("nfc |"));
-        for (uint8_t i = 0; i < 16; i++) {
-          Serial.print(mifareData[i] < 0x10 ? " 0" : " ");
-          Serial.print(mifareData[i], HEX);
-        }
-        Serial.println();
-
+      else
+      {
         // convert 4 byte cookie to 32bit decimal for easier handling
         uint32_t tempCookie;
-        tempCookie  = (uint32_t) mifareData[0] << 24;
-        tempCookie += (uint32_t) mifareData[1] << 16;
-        tempCookie += (uint32_t) mifareData[2] << 8;
-        tempCookie += (uint32_t) mifareData[3];
+        tempCookie = (uint32_t)mifareData[0] << 24;
+        tempCookie += (uint32_t)mifareData[1] << 16;
+        tempCookie += (uint32_t)mifareData[2] << 8;
+        tempCookie += (uint32_t)mifareData[3];
 
         // if cookie is not blank, update ncfTag object with data read from nfc tag
-        if (tempCookie != 0) {
+        if (tempCookie != 0)
+        {
           mfrc522.PICC_HaltA();
           mfrc522.PCD_StopCrypto1();
           nfcTag.cookie = tempCookie;
           nfcTag.version = mifareData[4];
           nfcTag.assignedFolder = mifareData[5];
           nfcTag.playbackMode = mifareData[6];
-        }
-        // if cookie is blank, clear ncfTag object
```

Comment:
You need to put this in again and actually add `nfcTag.track = 0;` at the end. Otherwise you can only learn a new tag when no other card triggered a playback before. After playing any card and then showing a blank card, actually just replays the same card and then locks up the reader, requiring a power cycle. Originally you can show a blank card at any time.

Reply:
Thank you so much for the review. Did not run into this bug yet, but it would have been a bummer. :-) :+1:

Reply:
I actually did run into this only a few weeks ago. I fixed it in b03e17d9437581a52542c3d07eb94bf7473cca73.